community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/StoreProcessor.java

```diff
@@ -145,4 +205,80 @@ public void processSchema( RecordStore<DynamicRecord> store, DynamicRecord schem
             checkSchema( RecordType.SCHEMA, store, schema, schemaRecordCheck );
         }
     }
+
+    public <R extends AbstractBaseRecord> void applyFilteredParallel( RecordStore<R> store,
+            Distributor<R> distributor, Predicate<? super R>... filters ) throws Exception
+    {
+        applyFilteredParallel( store, distributor, ProgressListener.NONE, filters );
+    }
+
+    public <R extends AbstractBaseRecord> void applyFilteredParallel( RecordStore<R> store,
+            Distributor<R> distributor, ProgressListener progressListener, Predicate<? super R>... filters )
+            throws Exception
+    {
+        applyParallel( store, distributor, progressListener, filters );
+    }
+
+    private <R extends AbstractBaseRecord> void applyParallel( RecordStore<R> store, Distributor<R> distributor,
+            ProgressListener progressListener, Predicate<? super R>... filters ) throws Exception
+    {
+        int threads = distributor.getNumThreads();
+        cacheAccess.prepareForProcessingOfSingleStore( distributor.getRecordsPerCPU() );
+        ArrayBlockingQueue<R>[] recordQ = new ArrayBlockingQueue[threads];
+        Workers<Worker<R>> workers = new Workers<>( getClass().getSimpleName() );
+        for ( int threadId = 0; threadId < threads; threadId++ )
+        {
+            recordQ[threadId] = new ArrayBlockingQueue<>( qSize );
+            workers.start( new Worker<>( recordQ[threadId], store ) );
+        }
+
+        int[] recsProcessed = new int[threads];
+        int[] qIndex = null;
+        int entryCount = 0;
+        Iterable<R> iterator = Scanner.scan( store, stage.isForward(), filters );
+        for ( R record : iterator )
+        {
+            try
+            {
+                qIndex = distributor.whichQ( record, qIndex );
+                for ( int i = 0; i < qIndex.length; i++ )
+                {
+                    while ( !recordQ[qIndex[i]].offer( record, 100, TimeUnit.MILLISECONDS ) )
+                    {
+                        qIndex[i] = (qIndex[i] + 1) % qIndex.length;
+                    }
+                    recsProcessed[qIndex[i]]++;
+                }
+            }
+            catch ( InterruptedException e )
+            {
+                Thread.currentThread().interrupt();
+                break;
+            }
+            progressListener.set( entryCount++ );
+        }
+        progressListener.done();
+        for ( Worker<R> worker : workers )
+        {
+            worker.done();
+        }
+        workers.awaitAndThrowOnError( RuntimeException.class );
```

Comment:
Simplified it quite a bit by removing the `Distributor` which wasn't really needed.